### PR TITLE
Japi 105 axis2 update

### PR DESCRIPTION
**Pull request by rpastrana**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [x] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [x] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by rpastrana** [comment]
@jpmcmu  this is still wip but please fetch locally and try to build  and look over the changes in org.hpccsystems.ws.client.axis2, and org.hpccsystems.ws.client.wrappers

File: commons-hpcc/src/main/java/org/hpccsystems/commons/commonclient.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+		    String authHeader = "Basic " + Base64.getEncoder().encodeToString(credentials.getBytes());
+		    headers.put(HTTPConstants.HEADER_AUTHORIZATION, authHeader);
+		    //stub._setProperty(HTTPConstants.REQUEST_HEADERS, headers);
+		    */
+			//WsDfuStub wsdfustub = new WsDfuStub(cc);
+			HttpTransportPropertiesImpl.Authenticator auth = new HttpTransportPropertiesImpl.Authenticator();
+			auth.setPreemptiveAuthentication(true);
+			auth.setUsername("rpastrana");
+			auth.setPassword("G0gator$");
+
```
Comment (drealeed) [inline commons-hpcc/src/main/java/org/hpccsystems/commons/commonclient.java]: cleartext password

**Comment by rpastrana** [comment]
@jpmcmu please review....
Please focus on wsclient\src\main\java\org\hpccsystems\ws\client\axis2\BaseHPCCWsClient.java and the classes that extend that class, also  org.hpccsystems.ws.client.wrappers.wsworkunits

File: wsclient/pom.xml [hunk 2]
```diff
@@ -16,6 +16,7 @@
     </licenses>
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <axis2.version>1.7.9</axis2.version>
```
Comment (rpastrana) [inline wsclient/pom.xml]: @Michael-Gardner please review this change

File: wsclient/src/main/java/org/hpccsystems/ws/client/axis2/HPCCWsSQLClient.java [hunk 3]
```diff
@@ -0,0 +1,571 @@
...
+        {
+            log.error("HPCCWsSQL.getTargetClusters("+filter+") encountered RemoteException.\n" + e.getLocalizedMessage());
+        }
+
+        handleEspExceptions(resp.getExceptions(), "HPCCWsSQL.getTargetClusters("+filter+") encountered RemoteException.");
+
+        return resp.getClusterNames().getClusterName();
+    }
+
+    public HPCCTable [] getTables(String filter) throws Exception
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/axis2/HPCCWsSQLClient.java]: HPCCTable needs a wrapper

File: wsclient/src/main/java/org/hpccsystems/ws/client/axis2/HPCCWsStoreClient.java [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+        }
+        catch (Exception e)
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    public String [] listNaspaces(String storename, boolean global) throws Exception
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/axis2/HPCCWsStoreClient.java]: typo in method name

File: wsclient/src/main/java/org/hpccsystems/ws/client/extended/axis2/HPCCWsAttributesClient.java [hunk 5]
```diff
@@ -0,0 +1,847 @@
...
+
+        if (modulename == null && attributename == null && username == null &&
+                type==null && anytext == null && changedSince == null)
+        {
+            throw new Exception("At least one find criteria is required.");
+        }
+
+        FindAttributes params = new FindAttributes();
+
+        if (modulename != null)
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/extended/axis2/HPCCWsAttributesClient.java]: We throw an exception above if these params are null. Should we remove the exception or remove the null checks?

Comment (rpastrana) [reply wsclient/src/main/java/org/hpccsystems/ws/client/extended/axis2/HPCCWsAttributesClient.java]: It looks like we throw the exception if all params are null, and then we avoid setting the individual param if it is null

File: wsclient/src/main/java/org/hpccsystems/ws/client/axis2/HPCCWsDFUClient.java [hunk 6]
```diff
@@ -0,0 +1,1566 @@
...
+                if (ex.getMessage() != null)
+                {
+                    errs = errs + ex.getMessage() + "\n";
+                }
+                log.error("Source: " + ex.getSource() + " Message: " + ex.getMessage());
+            }
+            throw new Exception(errs);
+        }
+    }
+    private void handleEspExceptions(EspSoapFault e, String message) throws Exception
```
[the remark this answers is not in the document]
Comment (rpastrana) [reply wsclient/src/main/java/org/hpccsystems/ws/client/axis2/HPCCWsDFUClient.java]: agree 100%, the problem is the EspSoapFault types are different, we'll need a EspSoapWrapper with a constructor for every generated EspSoapFault...

**Comment by rpastrana** [comment]
@jpmcmu please review latest changes

**Comment by rpastrana** [comment]
@jpmcmu please review latest changes

**Comment by rpastrana** [comment]
@drealeed please fetch this branch and test against your code.

There are significant changes in axis2/HPCCWsWorkUnitsClient.java none which should change our logic.

Except for this change which seemed to be a bug (please confirm)
https://github.com/hpcc-systems/HPCC-JAPIs/blob/898ecfb761b1e7366997501419289a3c31818d67/wsclient/src/main/java/org/hpccsystems/ws/client/axis2/HPCCWsWorkUnitsClient.java#L911

**Comment by rpastrana** [comment]
@drealeed please take a look at the latest commit which changes the way the clients handle esp exceptions.

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsClient.java [hunk 7]
```diff
@@ -471,6 +473,10 @@ public HPCCWsWorkUnitsClient getWsWorkunitsClient()
         {
             e.printStackTrace();
         }
+        catch (org.hpccsystems.ws.client.wrappers.ArrayOfEspExceptionWrapper e)
+        {
+            e.toString();
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsClient.java]: What are we doing with e.toString()? Should this be logged?

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java [hunk 8]
```diff
@@ -1458,7 +1484,7 @@ public DFUFileDetailWrapper getFileDetails(String logicalname, String clusternam
                         // for clusters on version < 5.0.2, this service doesn't exist; do it the old fashioned way
                         info.setColumns(this.getFileDataColumns(logicalname, clustername));
                     }
-                    rodrigo
+                    drea?
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java]: Do we still need these comments?

File: wsclient/src/main/java/org/hpccsystems/ws/client/platform/Workunit.java [hunk 9]
```diff
@@ -539,6 +540,10 @@ void fullRefresh(boolean includeGraphs, boolean includeResults, boolean includeS
                     }       
                 }
             }
+            catch (ArrayOfEspExceptionWrapper e)
+            {
+                System.out.println(e.toString());
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/platform/Workunit.java]: Should this be using log4j?

File: wsclient/src/test/java/org/hpccsystems/ws/client/WSSQLClientTest.java [hunk 10]
```diff
@@ -299,7 +326,11 @@ public void fullTest() throws Exception
             {
                 System.out.println("Exception while executing SQL: " + e.getLocalizedMessage());
             }
-
+            catch (ArrayOfEspExceptionWrapper e)
+            {
+                e.toString();
```
Comment (jpmcmu) [inline wsclient/src/test/java/org/hpccsystems/ws/client/WSSQLClientTest.java]: Should e.toString() be passed in as the cause to Assert.fail()?

File: wsclient/src/test/java/org/hpccsystems/ws/client/WSSQLClientTest.java [hunk 11]
```diff
@@ -380,5 +416,10 @@ public void fullTest() throws Exception
             e.printStackTrace();
             Assert.fail();
         }
+        catch (ArrayOfEspExceptionWrapper e)
+        {
+            e.toString();
+            Assert.fail();
```
Comment (jpmcmu) [inline wsclient/src/test/java/org/hpccsystems/ws/client/WSSQLClientTest.java]: Same question as above

File: wsclient/src/test/java/org/hpccsystems/ws/client/WSStoreClientTest.java [hunk 12]
```diff
@@ -123,6 +124,11 @@ public void listNamespaces()
         {
             Assert.fail(e.getLocalizedMessage());
         }
+        catch (ArrayOfEspExceptionWrapper e)
+        {
+            e.toString();
+            Assert.fail();
```
Comment (jpmcmu) [inline wsclient/src/test/java/org/hpccsystems/ws/client/WSStoreClientTest.java]: Same question as above

File: wsclient/src/test/java/org/hpccsystems/ws/client/WSWorkunitsTest.java [hunk 13]
```diff
@@ -84,6 +85,11 @@ public void fastWURefreshTest() throws Exception
             e.printStackTrace();
             Assert.fail();
         }
+        catch (ArrayOfEspExceptionWrapper e)
+        {
+            e.toString();
+            Assert.fail();
```
Comment (jpmcmu) [inline wsclient/src/test/java/org/hpccsystems/ws/client/WSWorkunitsTest.java]: Same question as above

File: wsclient/src/test/java/org/hpccsystems/ws/client/WsDFUClientTest.java [hunk 14]
```diff
@@ -133,6 +134,11 @@ public void recursiveGetFiles(String directory)
             e.printStackTrace();
             Assert.fail();
         }
+        catch (ArrayOfEspExceptionWrapper e)
+        {
+            e.toString();
```
Comment (jpmcmu) [inline wsclient/src/test/java/org/hpccsystems/ws/client/WsDFUClientTest.java]: Same question as above

**Pull request merged**